cpp/CMakeLists.txt

```diff
@@ -14,61 +14,95 @@ if(CCACHE_PROGRAM)
   message(STATUS "Using ccache: ${CCACHE_PROGRAM}")
 endif()
 
+option(USE_PACKAGE_MANAGER_DEPENDENCIES
+  "Try to find dependencies provided by the local package manager, and if not found, fall back to building them from source"
+  ON)
```

Comment:
Note to reviewers: if we agree on this approach, we can certainly discuss whether we want the default here to be `ON` or `OFF`.